### PR TITLE
ACMS-1137: Build own Acquia CMS Starter Kit.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1137

**Proposed changes**
Provide way to customer to build own starterkit.

**Alternatives considered**
NA

**Testing steps**
Follow from ticket.

### CODE REVIEW

File: src/Cli.php [hunk 1]
```diff
@@ -116,8 +116,10 @@ public function getAcquiaCmsFile(string $file_path): array {
    * Returns an array of starter-kits defined in acms.yml file.
    */
   public function getStarterKits(): array {
-    $fileContent = $this->getAcquiaCmsFile($this->projectDirectory . '/acms/acms.yml');
-    return $fileContent['starter_kits'] ?? [];
+    $defaultStarterkits = $this->getAcquiaCmsFile($this->projectDirectory . '/acms/acms.yml');
+    $userDefinedStarterkits = $this->getAcquiaCmsFile($this->rootDirectory . '/acms.yml');
+    $fileContent = array_merge($defaultStarterkits['starter_kits'] ?? [], $userDefinedStarterkits['starter_kits'] ?? []);
```
Comment (apathak18) [inline src/Cli.php:121]: No need of null coalescing operator.

Comment (apathak18) [reply src/Cli.php:121]: we can simplify line#121 with below statement.
$starterKits = array_merge($defaultStarterkits, $userDefinedStarterkits);

File: src/Cli.php [hunk 2]
```diff
@@ -116,8 +116,10 @@ public function getAcquiaCmsFile(string $file_path): array {
...
+    return $fileContent;
```
Comment (apathak18) [inline src/Cli.php:122]: Add one empty line above return statement and change return statement to return $starterKits['starter_kits'] ?? []

File: src/Cli.php [hunk 3]
```diff
@@ -116,8 +116,10 @@ public function getAcquiaCmsFile(string $file_path): array {
    * Returns an array of starter-kits defined in acms.yml file.
    */
   public function getStarterKits(): array {
-    $fileContent = $this->getAcquiaCmsFile($this->projectDirectory . '/acms/acms.yml');
-    return $fileContent['starter_kits'] ?? [];
+    $defaultStarterkits = $this->getAcquiaCmsFile($this->projectDirectory . '/acms/acms.yml');
+    $userDefinedStarterkits = $this->getAcquiaCmsFile($this->rootDirectory . '/acms.yml');
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply src/Cli.php:120]: @apathak18 have you tested this using testing steps to understand the difference?

File: src/Cli.php (hunk 1 above)
Comment (rajeshreeputra) [reply src/Cli.php:121]: @apathak18 this is not possible not possible as both has the starter_kits as parent key, hence only one will be returned, you can try by updating the same, at your end.

> we can simplify line#121 with below statement. $starterKits = array_merge($defaultStarterkits, $userDefinedStarterkits);

Comment (rajeshreeputra) [reply src/Cli.php:121]: When one of the file is missing it return nothing, which is not array and will return "Warning: Undefined array key 'starter_kits'", hence null coalescing operator is required.
> No need of null coalescing operator.

File: src/Cli.php (hunk 2 above)
Comment (rajeshreeputra) [reply src/Cli.php:125]: Added newline before return.

File: src/Cli.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (apathak18) [reply src/Cli.php:122]: Thanks clear now

File: src/Cli.php [hunk 4]
```diff
@@ -116,8 +118,11 @@ public function getAcquiaCmsFile(string $file_path): array {
...
```
Comment (apathak18) [inline src/Cli.php:122]: @rajeshreeputra Line#122 will throw exception in CLI if acms.yml file isn't exists at root directory, can we add a condition whether file exists or not? then do further logic implementation on the basis of that.

Comment (rajeshreeputra) [reply src/Cli.php:122]: it is already there see - https://github.com/acquia/acquia-cms-starterkit/pull/107/files#diff-2715e1365c87080f44dc9848449f6101a079ed8a60928677f487d52f41ec661eR107-R109

Comment (apathak18) [reply src/Cli.php:122]: Screenshot 2023-07-21 at 9 39 47 AM

File: src/Cli.php (hunk 1 above)
Comment (rajeshreeputra) [reply src/Cli.php:123]: @apathak18 also update logic to remove the null coalescing operator, thanks!! 😊

**Review by apathak18 — APPROVED** [review]

**Review by deepakmishra2 — APPROVED** [review]
Looks good to me

**Review by chandan-singh7929 — APPROVED** [review]

**Review by vishalkhode1 — APPROVED** [review]

File: src/Cli.php [hunk 5]
```diff
@@ -109,15 +109,29 @@ public function getAcquiaCmsFile(string $file_path): array {
...
    */
   public function getStarterKits(): array {
-    $fileContent = $this->getAcquiaCmsFile($this->projectDirectory . '/acms/acms.yml');
-    return $fileContent['starter_kits'] ?? [];
+    $starterkits = [];
+    $defaultStarterkits = $this->getAcquiaCmsFile($this->projectDirectory . '/acms/acms.yml');
+    $starterkits = $defaultStarterkits['starter_kits'];
+    // Check if user defined starterkit file exist in root directory.
+    if ($this->filesystem->exists($this->rootDirectory . '/acms.yml')) {
+      $userDefinedStarterkits = $this->getAcquiaCmsFile($this->rootDirectory . '/acms.yml');
```
Comment (vishalkhode1) [inline src/Cli.php:126]: @rajeshreeputra We should look for file in `acms` folder of root directory.

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: src/Cli.php (hunk 5 above)
Comment (rajeshreeputra) [reply src/Cli.php:126]: As discussed will update location from root directory to root/acms/acms.yml.

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**